pkg/utils/vault/secret_file.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package vault
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	// inMemory is the name of the file to use to not write it to disk and only keep it in memory
+	inMemory = "in-memory"
```

Comment:
Feels like maybe this should be exported so someone outside of this package can `SecretFile{Name: vault.InMemoryFile}` or something?

Or maybe have `func NewSecretFile` and `func NewInMemorySecretfile` with `inMemory` just being `bool`?

Reply:
Yes, we can export it but currently it is not used in other place. We want to keep the flexibility and be able to choose between the 3 modes (don't read, read in mem or read and write) depending on the context:
- in ci: `MY_SECRET_FILE=""` (pr) || `MY_SECRET_FILE=in-memory`
- dev: `MY_SECRET_FILE=""` || `MY_SECRET_FILE=/a/path/to/a/file (to reproduce ci but not get the secret at each try)